### ISSUE TITLE
Make it easier to preview themes

### ISSUE THREAD

**Issue by jenlampton** [backdrop-ops/backdropcms.org#762]
We need a better way for people to choose a theme. This includes seeing what the theme will look like on phones and tablets as well as on desktop.

This could be done using individual TugBoat sandboxes, a single site that has all themes installed, or some kind of on-demand set-up that adds the theme somewhere for you to view it.

We've also discussed having a complete site (containing content, layouts, and views) rather testing these themes on a "blank slate" site. This would allow people to make a more informed decision.

**Comment by stpaultim** [comment]
Here is a tool that I've been working on to help folks better test themes. This does not address all the issues that you mention and does not use the users end content. 

However this is a site with some sample content and the ability to switch from theme to theme. Is this a helpful step in the right direction. 

On a related note, this is the site that @cellear has been working on for Ready2Wear initiative. 
http://ready2wear.simplo.site/ (still in development).

**Comment by docwilmot** [comment]
Would also be great if we could allow theme devs to include **full sized screenshots** on backdropcms.org. 
Look here: https://github.com/backdrop-ops/backdropcms.org/issues/487

**Comment by BWPanda** [comment]
One of the sites I've seen that provides live demos of its themes is ThemeForest: https://themeforest.net/category/site-templates

Its demos are separate, individual sites run by the creators of the themes. So if I added a theme there, I could link to a demo I've setup of that theme at, for example, theme-demo.panda.id.au. in other words, ThemeForest itself doesn't create, manage or host the demo sites.

Backdrop theme developers can already do this - create and host a site that showcases their theme. And I think this is probably the best way to do it, as each theme needs its own specific configuration, content and layout(s) to showcase what it can do the best. I don't think a one-size-fits-all solution would work here. Consider the needs of a basic front-end theme vs. an advanced back-end/admin theme.

**Comment by stpaultim** [comment]
> Backdrop theme developers can already do this - create and host a site that showcases their theme. And I think this is probably the best way to do it, as each theme needs its own specific configuration, content and layout(s) to showcase what it can do the best.

Yes, this is all true. But, how many theme developers have done this? I am unaware of any demo sites for any of our existing themes.

We frequently get feedback from folks testing Backdrop CMS that they are frustrated by the lack of themes. Making it easier to see and test themes, might reduce this concern. I think it's in the interest of the community that we make it easy to discover and view themes, letting theme developers do it on their own is not working very well - even if in theory it's the best option. 

There are things we can do as a community here:

1) We could do better at encouraging theme developers to create demo sites.
2) We could make it easier for users to SPIN up a site using Tugboat Demo site with sample content and a theme they would like to try. I recognize that this is already possible, but I highly suspect that most people don't understand that and wouldn't even think to try it. I think it would be realistic and reasonable to make this easier do do and/or easier to understand.
3) We could do better at encouraging theme developers to submit screenshots of their themes in action for showcase on BackdropCMS.org.
4) While it's clearly not the ideal answer, a site like https://themeswitcher.simplo.site/ is in my opinion better than nothing and I certainly could do more to make it even better or more useful. 

The lack of a one size fits all solution does not mean that there is nothing we can do. It might just mean that we'll need to adopt a few strategies and/or be more assertive in encouraging theme developers to create demo sites. 

I agree that there is not quick fix here and this may not be the most urgent problem to address, but I think it's something to keep thinking about and experimenting with.

**Comment by docwilmot** [comment]
Also, for a few months now we have the ability to allow unlimited full-sized screenshots to themes, that will be saved to the project nodes. We havent however, developed a UI to allow theme-seekers to browse those screenshots.

**Comment by BWPanda** [comment]
Ok, after looking at the https://themeswitcher.simplo.site/ site properly, I have some thoughts that I'll put below. Some of my thoughts are for the contrib modules used on the site, but I'll brain-dump everything here first then we can look at creating individual issues as necessary.

I think with some work this could be a good solution for most front-end themes (which is most themes anyway), and for those few themes that need something a bit different (like an admin theme), a dedicated solution is always still an option for them.

Thoughts:
- Site could be moved to something like https://themes.backdropcms.org so that it's more official and multiple contributors could be given admin access to it (I don't think it needs to be put into a repo, etc.; we can just make changes to the config/content directly as needed).
- Theme switcher block should be moved to the header for easier use (so you don't have to scroll down long pages to change themes)
- Theme switcher should work without needing a 'Switch' button (e.g. change themes when selecting an item from the list)
- The Styleguide should have a link in the main menu - I think this link should be dynamic, in that it'll always take you to the styleguide page of the theme currently in use (e.g. when viewing the Mero theme, the Styleguide link takes you to `/admin/appearance/styleguide/mero`)

(I'll update this list with more if/as I think of them)

**Comment by BWPanda** [comment]
@stpaultim I've made a PR to the Sample Animal Content repo to clean it up a bit and fix some bugs: https://github.com/stpaultim/sample_animal_content/issues/1
